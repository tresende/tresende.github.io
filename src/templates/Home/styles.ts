import styled from 'styled-components'

export const Main = styled.main`
  height: 100%;
  width: 100%;
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.6);
  @media only screen and (max-width: 760px) {
    background-color: rgba(0, 0, 0, 0.7);
  }
`

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  background-color: rgba(0, 0, 0, 0.6);
  color: white;
  max-width: 40rem;
  padding: 4rem;
  @media only screen and (max-width: 760px) {
    height: 100vh;
    padding: 2rem;
  }
`
