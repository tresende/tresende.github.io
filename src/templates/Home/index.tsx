import Head from 'next/head'

import * as S from './styles'

export default function Home() {
  return (
    <>
      <Head>
        <title>(2) Whatsapp</title>
        <link rel="icon" href="/favicon.ico" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
      </Head>
      <S.Video preload="none" autoPlay muted loop>
        <source
          src="https://raw.githubusercontent.com/tresende/tresende.github.io/master/public/videos/bg.mp4"
          type="video/mp4"
        />
      </S.Video>
      <S.Main>
        <S.Container className="flex">
          <S.Terminal>
            <S.Path>
              ~/Projects/tresende.github.io <S.Branch>main*</S.Branch>
            </S.Path>
            <S.Cursor>‚ùØ </S.Cursor>
            <span>cat readme.txt</span>
            <S.Block>_</S.Block>
          </S.Terminal>
          <S.Bio>
            Hello, friend.
            <br />
            My name is Thiago Resende. I&apos;m a frontend developer living in
            Belo Horizonte, Brazil. I&apos;m a fan of technology, programming,
            and coffee. These are my high-level skills: <br />
            - React <br />
            - Angular <br />
            - Node <br />
            - Flutter <br />
            - SNES <br />- Play bass
          </S.Bio>
          <S.SocialMedia>
            <a
              href="https://github.com/tresende"
              rel="noreferrer"
              target="_blank"
            >
              <img src="/img/github.png" alt="Github icon" />
            </a>
            <a
              href="https://www.linkedin.com/in/thiago-resende-46175391"
              rel="noreferrer"
              target="_blank"
            >
              <img src="/img/linkedin.png" alt="Linkedin icon" />
            </a>
            <a
              href="mailto:thiago.gcresende@gmail.com"
              target="_blank"
              rel="noreferrer"
            >
              <img src="/img/email.png" alt="Mail icon" />
            </a>
            <a
              rel="noreferrer"
              href="https://www.facebook.com/thiagooooooooooooooo"
              target="_blank"
            >
              <img src="/img/facebook.png" alt="Facebook icon" />
            </a>
            <a
              href="https://www.instagram.com/thiago.gcr/"
              target="_blank"
              rel="noreferrer"
            >
              <img src="/img/instragram.png" alt="Instagram icon" />
            </a>
          </S.SocialMedia>
        </S.Container>
      </S.Main>
    </>
  )
}
