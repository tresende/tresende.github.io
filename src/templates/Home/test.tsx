import { render } from '@testing-library/react'

import Home from '.'

describe('<Home />', () => {
  it('should render all components', () => {
    const { container } = render(<Home />)

    expect(container).toMatchSnapshot()
  })
})
