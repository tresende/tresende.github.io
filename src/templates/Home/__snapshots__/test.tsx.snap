// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Home /> > should render all components 1`] = `
<div>
  <main
    class="container relative mx-auto scroll-my-12 overflow-auto p-4 print:p-12 md:p-16"
  >
    <section
      class="mx-auto w-full max-w-2xl space-y-8 print:space-y-6"
    >
      <div
        class="space-y-1.5"
      >
        <h1
          class="text-2xl font-bold"
        >
          Thiago Resende
        </h1>
        <p
          class="max-w-md text-pretty font-mono text-sm text-muted-foreground"
        >
          Software engineer focused on building amazing products
        </p>
        <p
          class="max-w-md items-center text-pretty font-mono text-xs text-muted-foreground"
        >
          <a
            class="inline-flex gap-x-1.5 align-baseline leading-none hover:underline"
            href="https://www.google.com/maps/place/Belo+Horizonte,+State+of+Minas+Gerais"
            rel="noreferrer"
            target="_blank"
          >
            <span>
              üåè Belo Horizonte, MG, Brazil
            </span>
          </a>
        </p>
        <nav
          class="flex gap-x-1 pt-1 font-mono text-sm text-muted-foreground print:hidden"
        >
          <a
            aria-label="Email"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground size-8"
            href="mailto:thiago.gcresende@gmail.com"
          >
            <svg
              class="lucide lucide-mail size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="16"
                rx="2"
                width="20"
                x="2"
                y="4"
              />
              <path
                d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
              />
            </svg>
          </a>
          <a
            aria-label="Github"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground size-8"
            href="https://github.com/tresende"
          >
            <svg
              class="size-4"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="LinkedIn"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground size-8"
            href="https://www.linkedin.com/in/thiago-resende-46175391"
          >
            <svg
              class="size-4"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                LinkedIn
              </title>
              <path
                d="M20.447 20.452h-3.554v-5.569c0-1.328-.027-3.037-1.852-3.037-1.853 0-2.136 1.445-2.136 2.939v5.667H9.351V9h3.414v1.561h.046c.477-.9 1.637-1.85 3.37-1.85 3.601 0 4.267 2.37 4.267 5.455v6.286zM5.337 7.433c-1.144 0-2.063-.926-2.063-2.065 0-1.138.92-2.063 2.063-2.063 1.14 0 2.064.925 2.064 2.063 0 1.139-.925 2.065-2.064 2.065zm1.782 13.019H3.555V9h3.564v11.452zM22.225 0H1.771C.792 0 0 .774 0 1.729v20.542C0 23.227.792 24 1.771 24h20.451C23.2 24 24 23.227 24 22.271V1.729C24 .774 23.2 0 22.222 0h.003z"
                fill="currentColor"
              />
            </svg>
          </a>
        </nav>
      </div>
      <section
        class="flex min-h-0 flex-col gap-y-3"
      >
        <h2
          class="text-xl font-bold"
        >
          About
        </h2>
        <p
          class="text-pretty font-mono text-sm text-muted-foreground"
        >
          With over 
          14
           years of software development experience, I am a senior software engineer passionate about creating innovative and highly scalable solutions. My goal is to leverage my software engineering skills to deliver high-quality, high-performance, and easy-to-use products.
        </p>
        <p
          class="text-pretty font-mono text-sm text-muted-foreground"
        >
          I am currently working on a high-traffic application using Java, Kotlin, and microservices. This project involves developing and maintaining a robust system that supports extensive logistics operations. My work focuses on ensuring the application's scalability, reliability, and efficiency to meet the demands of millions of users.
        </p>
        <p
          class="text-pretty font-mono text-sm text-muted-foreground"
        >
          I also strive to contribute to a culture of continuous learning, collaboration, and diversity within my team by sharing my experiences, ideas, and feedback.
        </p>
      </section>
      <section
        class="flex min-h-0 flex-col gap-y-3"
      >
        <h2
          class="text-xl font-bold"
        >
          Work Experience
        </h2>
        <div
          class="rounded-lg bg-card text-card-foreground"
        >
          <div
            class="flex flex-col space-y-1.5"
          >
            <div
              class="flex items-center justify-between gap-x-2 text-base"
            >
              <h3
                class="inline-flex items-center justify-center gap-x-1 font-semibold leading-none"
              >
                <a
                  class="hover:underline"
                  href="https://ifood.com.br/"
                >
                  iFood
                </a>
                <span
                  class="inline-flex gap-x-1"
                />
              </h3>
              <div
                class="text-sm tabular-nums text-gray-500"
              >
                2021 - now
              </div>
            </div>
            <h4
              class="font-mono text-sm leading-none"
            >
              Senior Software Engineer
            </h4>
          </div>
          <div
            class="text-pretty font-mono text-muted-foreground mt-2 text-xs"
          >
            I am currently working on a web portal developed to help all logistics involved in iFood operations around the world. This application is developed using Java, Kotlin, microservices, TypeScript, Webpack, Testing Library, aws, and nx monorepo.
          </div>
        </div>
        <hr
          class="border-gray-200 dark:border-white"
        />
        <div
          class="rounded-lg bg-card text-card-foreground"
        >
          <div
            class="flex flex-col space-y-1.5"
          >
            <div
              class="flex items-center justify-between gap-x-2 text-base"
            >
              <h3
                class="inline-flex items-center justify-center gap-x-1 font-semibold leading-none"
              >
                <a
                  class="hover:underline"
                  href="https://hotmart.com"
                >
                  Hotmart
                </a>
                <span
                  class="inline-flex gap-x-1"
                />
              </h3>
              <div
                class="text-sm tabular-nums text-gray-500"
              >
                2020 - 2021
              </div>
            </div>
            <h4
              class="font-mono text-sm leading-none"
            >
              Senior Software Engineer
            </h4>
          </div>
          <div
            class="text-pretty font-mono text-muted-foreground mt-2 text-xs"
          >
            During my time at Hotmart, I worked as a senior frontend developer in two large projects. The first was focused on user privacy, creating internal applications in React with TypeScript and features in existing products for end users, using Next, Meteor, and Vue.js to meet all demands of legislation such as GDPR, LGPD, cookie policy, etc. In the second project, I worked on a product vital for Hotmart business. Our stack included React, Redux, Cypress, Vitest, and Meteor.
          </div>
        </div>
        <hr
          class="border-gray-200 dark:border-white"
        />
        <div
          class="rounded-lg bg-card text-card-foreground"
        >
          <div
            class="flex flex-col space-y-1.5"
          >
            <div
              class="flex items-center justify-between gap-x-2 text-base"
            >
              <h3
                class="inline-flex items-center justify-center gap-x-1 font-semibold leading-none"
              >
                <a
                  class="hover:underline"
                  href="https://www.dtidigital.com.br/"
                >
                  dti digital
                </a>
                <span
                  class="inline-flex gap-x-1"
                />
              </h3>
              <div
                class="text-sm tabular-nums text-gray-500"
              >
                2019 - 2020
              </div>
            </div>
            <h4
              class="font-mono text-sm leading-none"
            >
              Senior Software Engineer
            </h4>
          </div>
          <div
            class="text-pretty font-mono text-muted-foreground mt-2 text-xs"
          >
            At the beginning of my journey at DTI, I worked as a developer using .NET Core, Angular, and SQL Server. Soon after, I took on the role of technical leader for squads aimed at Vale, one of the largest mining companies in the world. We worked with a mix of recent technologies such as .NET Core (serverless), Protocol Buffers, Redis, Cosmos DB, and micro frontends (React and Angular) alongside traditional technologies like Java, SQL Server, and Power BI. We focused heavily on agile methodologies and used Scrum with adaptations for the client reality.
          </div>
        </div>
        <hr
          class="border-gray-200 dark:border-white"
        />
        <div
          class="rounded-lg bg-card text-card-foreground"
        >
          <div
            class="flex flex-col space-y-1.5"
          >
            <div
              class="flex items-center justify-between gap-x-2 text-base"
            >
              <h3
                class="inline-flex items-center justify-center gap-x-1 font-semibold leading-none"
              >
                <a
                  class="hover:underline"
                  href="https://ibm.com/"
                >
                  IBM
                </a>
                <span
                  class="inline-flex gap-x-1"
                />
              </h3>
              <div
                class="text-sm tabular-nums text-gray-500"
              >
                2015 - 2019
              </div>
            </div>
            <h4
              class="font-mono text-sm leading-none"
            >
              Full Stack Developer
            </h4>
          </div>
          <div
            class="text-pretty font-mono text-muted-foreground mt-2 text-xs"
          >
            During my time at IBM, I worked for two major players in the Brazilian automotive industry, FCA/Stellantis and Volkswagen. Initially, I worked as a full stack developer creating applications to improve supply chain management processes. These systems were mostly built in Java and Node.js on the backend and Angular and React on the frontend. Over time, I transitioned to software architecture, helping migrate and update these applications from FIAT internal data centers to AWS, initially using Beanstalk, Docker, and later Kubernetes. At Volkswagen, I developed a chatbot orchestrator to assist users with various internal processes. This application was built using Node.js, MongoDB, IBM Watson, and Angular, and was hosted on the IBM Cloud.
          </div>
        </div>
        <hr
          class="border-gray-200 dark:border-white"
        />
        <div
          class="rounded-lg bg-card text-card-foreground blur-sm"
        >
          <div
            class="flex flex-col space-y-1.5"
          >
            <div
              class="flex items-center justify-between gap-x-2 text-base"
            >
              <h3
                class="inline-flex items-center justify-center gap-x-1 font-semibold leading-none"
              >
                <a
                  href="https://www.linkedin.com/in/thiago-resende-46175391"
                  rel="noreferrer"
                  target="_blank"
                >
                  Why so curious?
                </a>
                <span
                  class="inline-flex gap-x-1"
                />
              </h3>
              <div
                class="text-sm tabular-nums text-gray-500"
              >
                <a
                  href="https://www.linkedin.com/in/thiago-resende-46175391"
                  rel="noreferrer"
                  target="_blank"
                >
                  20?? - 20??
                </a>
              </div>
            </div>
            <h4
              class="font-mono text-sm leading-none"
            >
              <a
                href="https://www.linkedin.com/in/thiago-resende-46175391"
                rel="noreferrer"
                target="_blank"
              >
                Nope
              </a>
            </h4>
          </div>
          <div
            class="text-pretty font-mono text-muted-foreground mt-2 text-xs"
          >
            <a
              href="https://www.linkedin.com/in/thiago-resende-46175391"
              rel="noreferrer"
              target="_blank"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed scelerisque ligula nec accumsan imperdiet. Praesent imperdiet ut nibh quis pellentesque. Morbi cursus ut urna in euismod. Curabitur vulputate sagittis est, vel venenatis nunc efficitur et. Sed pretium nisl et enim tempor, ut pharetra augue porta. Vestibulum scelerisque tortor magna, vel cursus turpis faucibus dapibus. Phasellus vitae dictum urna, a condimentum lorem. Cras vehicula laoreet justo, nec commodo enim viverra a.
            </a>
          </div>
        </div>
      </section>
      <section
        class="flex min-h-0 flex-col gap-y-3"
      >
        <h2
          class="text-xl font-bold"
        >
          Skills
        </h2>
        <ul
          class="flex flex-wrap gap-1"
        >
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            TypeScript
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            Node.js
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            Java
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            Kotlin
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            AWS
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            Microservices
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            SQL/NoSQL
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            K8S
          </li>
          <li
            class="inline-flex items-center rounded-md border px-2 py-0.5 text-xs font-semibold font-mono transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-nowrap border-transparent bg-primary/80 text-primary-foreground hover:bg-primary/60"
          >
            React
          </li>
        </ul>
      </section>
    </section>
  </main>
</div>
`;
